io/io/src/RRawFile.cxx

```diff
@@ -173,14 +147,20 @@ size_t ROOT::Internal::RRawFile::ReadAt(void *buffer, size_t nbytes, std::uint64
    EnsureOpen();
    R__ASSERT(fOptions.fBlockSize >= 0);
 
-   // "Large" reads are served directly, bypassing the cache
+   // Early return for empty requests
+   if (nbytes == 0)
+      return 0;
```

Comment:
Just curious: do we have empty requests in practice?

Reply:
No, I didn't observe it in practice, I just noticed that in case `fOptions.fBlockSize` is also empty the code was taking unexpected paths into the buffered I/O case